src/docs/patterns/overlay.mdx

```diff
@@ -1,73 +1,132 @@
+# Variations
 
-Modal dialogs are also very commonly used to confirm a user action (such as deleting an item). These usually appear in the center of the screen.
+Overlay elements come in a variety of shapes and sizes depending on their purpose and can be either blocking (i.e., the user must interact with the overlay before they can do anything else) or non-blocking.
+
+## Modal Dialog
 
-<MaterialDesignDescription
-    title={'Dialogs'}
-    url={'https://material.io/components/dialogs/'}
-    description={'Read more about dialogs on Material.io.'}
+<ImageGrid
+    images={[DesktopDialog, DesktopDialogTask]}
+    caption={
+        'A modal dialog appears centered in the screen over a dark overlay. These can present information to the user, or prompt them to enter information.'
+    }
 />
+<MaterialDesignDescription title={'Dialogs'} url={'https://material.io/components/dialogs/'} />
+
+The modal dialog is your typical pop-up window that appears in the center of the screen in front of primary content. The primary use of dialogs is to present critical information to users or to ask for a decision. For example, you may wish to ask a user if they are sure they want to delete a file.
+
+Modals are also great for presenting single tasks or workflows to a user. By using a modal, you allow the user to do more work without cluttering your main layout with unnecessary information (e.g., an Add User button could open a modal with a form to enter the new user's information). You can also use modals to edit large amounts of information that would be too difficult to fit into a sidebar.
+
+Modal dialogs are always blocking — they must be dismissed before the user can interact with the rest of the application content. They may be dismissed by clicking on a close button or, for less critical dialogs, by clicking anywhere on the screen outside of it.
 
 ## Sidebar
 
-The most common example of a sidebar is the application navigation menu. Typically, this opens on the side of the screen and is used to navigate to the different areas of the application.
+<MaterialDesignDescription title={'Sheets: Side'} url={'https://material.io/components/sheets-side'} />
 
-Sidebars are also used extensively for displaying information about selected items in the main interface (e.g., selecting an item in a list would open a sidebar on the right with additional details about the selected item). They can also present simple form data and inputs (such as radio buttons, checkbox, dropdown menus, etc.) for editing a selected item. This type of sidebar should be temporary such that the user can easily close it with a close button or clicking elsewhere on the screen.
+Sidebars (Side Sheets in the Material documentation) are used to present supplemental information to users. Most commonly, they display information about selected items in the main interface (e.g., selecting an item in a list would open a sidebar on the right with additional details about the selected item).
```

Comment:
In addition to selected items, there _might_ be cases where we have one big item occupying the entire screen (like a map), and a permanent side sheet (like a navigator / inspector of the map).